### PR TITLE
Simplifying Selection Tiers

### PR DESCRIPTION
**Pull request by asfreitas**
This is in regard to https://github.com/OpenRA/OpenRA/issues/16555

Simplifies the way selection tiers works and changes some of them so that selections work in a way that further prioritizes combat units and buildings.

### CODE REVIEW

File: mods/ra/rules/defaults.yaml [hunk 1]
```diff
@@ -626,17 +626,16 @@
 		RequiresCondition: cruising
 	BodyOrientation:
 		UseClassicFacingFudge: True
-
+
```
Comment (abcdefg30) [inline mods/ra/rules/defaults.yaml]: Please revert this change.

File: mods/ra/rules/defaults.yaml [hunk 2]
```diff
@@ -651,7 +650,7 @@
 	SoundOnDamageTransition:
 		DamagedSounds: kaboom1.aud
 		DestroyedSounds: kaboom22.aud
-	WithSpriteBody:
+	WithSpriteBody:
```
Comment (abcdefg30) [inline mods/ra/rules/defaults.yaml]: This too.

File: mods/ra/rules/infantry.yaml [hunk 3]
```diff
@@ -193,7 +193,7 @@ E3:
 		Prerequisites: barracks.upgraded
 	AutoTarget:
 		ScanRadius: 5
-
+
```
Comment (abcdefg30) [inline mods/ra/rules/infantry.yaml]: This as well.

File: mods/ra/rules/structures.yaml [hunk 4]
```diff
@@ -666,7 +666,7 @@ DOME:
 	ProvidesPrerequisite@buildingname:
 	ExternalCondition@JAMMED:
 		Condition: jammed
-
+
```
Comment (abcdefg30) [inline mods/ra/rules/structures.yaml]: Same here.

File: mods/ra/rules/vehicles.yaml [hunk 5]
```diff
@@ -286,9 +286,10 @@ ARTY:
 		LoadedChance: 75
 	ProducibleWithLevel:
 		Prerequisites: vehicles.upgraded
-
+
```
Comment (abcdefg30) [inline mods/ra/rules/vehicles.yaml]: And here.

File: mods/ra/rules/defaults.yaml [hunk 6]
```diff
@@ -1212,3 +1212,24 @@
 		Types: SpyInfiltrate
 	Power:
 		RequiresCondition: !disabled
+
+
```
Comment (abcdefg30) [inline mods/ra/rules/defaults.yaml]: Duplicate new line.

File: mods/ra/rules/defaults.yaml [hunk 7]
```diff
@@ -626,17 +626,16 @@
...
 		UseClassicFacingFudge: True
-
+	
 ^BasicBuilding:
 	Inherits@1: ^ExistsInWorld
 	Inherits@2: ^IronCurtainable
 	Inherits@3: ^SpriteActor
 	Inherits@shape: ^1x1Shape
 	Inherits@bounty: ^GlobalBounty
+	Inherits@SELECTION_TIER: ^SelectableBuilding
```
Comment (abcdefg30) [inline mods/ra/rules/defaults.yaml]: `SELECTION_TIER` doesn't seem to fit the other suffices. What about just using `@selection`?

**Review by abcdefg30 — COMMENTED** [review]
This looks like a good start to me. A few style nits:

File: mods/ra/rules/defaults.yaml (hunk 7 above)
Comment (dragunoff) [reply mods/ra/rules/defaults.yaml]: That name was suggested by me, because I am proposing in #15839 a `Inherits@SELECTION_MODE: ^FlatSelectionMode`. Though these templates could be combined into a single template.

Comment (pchote) [reply mods/ra/rules/defaults.yaml]: I believe @abcdefg30 meant the `SELECTION_TIER` name, vs `1`, `shape`, `bounty`, etc.

Comment (dragunoff) [reply mods/ra/rules/defaults.yaml]: That's what I meant as well :-) We can combine `SELECTION_MODE` and `SELECTION_TIER` into the same declaration (if the other PR gets merged). So we'd then have for example:
```
^SelectableSupportUnit:
	Selectable:
		Priority: 8
		PriorityModifiers: Ctrl, Alt
```
```
	Inherit@SELECTION: ^SelectableSupportUnit
```

Comment (abcdefg30) [reply mods/ra/rules/defaults.yaml]: I'm just asking about the upper case and the underscore - not common standard afaik.

Comment (asfreitas) [reply mods/ra/rules/defaults.yaml]: I'll make these changes over the weekend. Thanks for your feedback!

Comment (dragunoff) [reply mods/ra/rules/defaults.yaml]: I thought that uppercase-underscore is the convention in OpenRA for such cases 🤔 it certainly is in chrome yamls.

Comment (abcdefg30) [reply mods/ra/rules/defaults.yaml]: Yes, those are different. ^^)

**Comment by pchote** [comment]
Superseded by #16662.
